python/pyspark/sql/datasource.py

```diff
@@ -469,6 +501,188 @@ def stop(self) -> None:
         ...
 
 
+class SimpleInputPartition(InputPartition):
+    def __init__(self, start: dict, end: dict):
+        self.start = start
+        self.end = end
+
+
+class SimpleDataSourceStreamReader(ABC):
+    """
+    A base class for simplified streaming data source readers. Compared to DataSourceStreamReader,
+    SimpleDataSourceStreamReader doesn't require planning data partitioning. Also, the read api of
```

Comment:
I would use directives to reference classes here

```suggestion
    :class:`SimpleDataSourceStreamReader` doesn't require planning data partitioning. Also, the read api of
```